pipelines/test/staticpackage.yaml

```diff
@@ -0,0 +1,33 @@
+name: staticpackage
+
+# static packages that contain only static libraries, which are linked into applications at compile time rather than at runtime.
+# this packages only contains `.a` files, no other files are expected. this pipeline checks that the package contains only static libraries.
+pipeline:
+  - name: static package check
+    runs: |
+      # Get our static package name, which is usually a sub package
+      static_pkg=$(basename ${{targets.contextdir}})
+
+      # Get file list, excluding empty lines and SBOM files
+      file_list="$(apk info --quiet --contents "$static_pkg" | grep -v "^$" | grep -v "^var/lib/db/sbom")"
+
+      # Check if package contains any .a files
+      if [ $(echo "$file_list" | grep "\.a$" | wc -l) -eq 0 ]; then
+        echo "See:"
+        apk info -qL "$static_pkg"
+        echo "FAIL: Package [$static_pkg] does not contain any static library '.a' files"
+        echo "      Please check the package build for proper static library installation, and either:"
+        echo "        (a) fix the static subpackage build to actually include .a files (check the split/static pipeline), or"
+        echo "        (b) remove the test/staticpackage pipeline"
+        exit 1
+      fi
+
+      # Check if package contains any non-.a files
+      if [ $(echo "$file_list" | grep -v "\.a$" | wc -l) -ne 0 ]; then
+        echo "FAIL: Package [$static_pkg] contains unexpected files beyond static library '.a' files:"
+        echo "$file_list" | grep -v "\.a$"
+        echo "      Expected only .a files and .spdx.json metadata"
+        exit 1
+      fi
+
+      echo "PASS: Package [$static_pkg] contains only valid static libraries, total static files: $static_files"
```

Comment:
this `$static_files` var is unset.

Reply:
@smoser After reviewing tw and thinking it through, I believe this test fits better alongside the existing tests for metapackages, virtual packages, empty packages, and documentation packages.

So I think i would like to have this test pipeline here instead of tw.

Reply:
ultimately, i'd like to get those other things to tw also.

tw allows to more easily test and improve pipelines and tools that are used in pipelines.

We can be more consistent on output and pipeline behavior.

We can write tests for the pipelines, and use things like shellcheck, or replace bash with a more suitable programming language with ease.

Reply:
> ultimately, i'd like to get those other things to tw also.
> 
> tw allows to more easily test and improve pipelines and tools that are used in pipelines.
> 
> We can be more consistent on output and pipeline behavior.
> 
> We can write tests for the pipelines, and use things like shellcheck, or replace bash with a more suitable programming language with ease.

I would be happy to prepare a PR for that repo.

I will send you a message via slack to learn a bit more how the tw repo is made available to the other 3 melange pacakge repos.